core/frontend/src/components/notifications/NotificationManager.vue

```diff
@@ -0,0 +1,135 @@
+      <v-spacer />
+      <v-btn
+        icon
+        @click="dismissAllNotifications()"
+      >
+        <v-icon>mdi-broom</v-icon>
+      </v-btn>
+      <v-menu
+        v-model="show_config_menu"
+        :close-on-content-click="false"
+        :nudge-x="400"
+        offset-y
+      >
+        <template #activator="{on, attrs}">
+          <v-btn
+            icon
+            v-bind="attrs"
+            v-on="on"
+          >
+            <v-icon>mdi-cog</v-icon>
+          </v-btn>
+        </template>
+        <notifications-config-menu
+          :show-old.sync="show_old_messages"
+        />
+      </v-menu>
+    </v-app-bar>
+
+    <v-sheet
+      id="notifications-list"
+      class="overflow-y-auto"
+    >
+      <v-list
+        three-line
+        dense
+        class="overflow-y-auto"
+        tile
+        max-height="50vh"
+      >
+        <v-list-item-group
+          v-if="notifications_to_show.length != 0"
+        >
+          <notification-card
+            v-for="({notification, count}, index) in notifications_to_show"
+            :key="index"
+            :notification="notification"
+            :count="count"
+            :timestamp="timestamp"
+            @dismiss="dismissNotification"
+          />
+        </v-list-item-group>
+
+        <v-card
+          v-else
+          elevation="0"
+        >
+          <v-card-text class="text-center">
+            No notifications to display :)
+          </v-card-text>
+        </v-card>
+      </v-list>
+    </v-sheet>
+  </v-card>
+</template>
+
+<script lang="ts">
+import { CumulatedNotification, Notification } from '@/types/notifications'
+import NotificationCard from './NotificationCard.vue'
+import NotificationStore from '@/store/notifications'
+import NotificationsConfigMenu from './ConfigMenu.vue'
+import Vue from 'vue'
+import { getModule } from 'vuex-module-decorators'
+
+const notification_store: NotificationStore = getModule(NotificationStore)
+
+export default Vue.extend({
+  name: 'NotificationManager',
+  components: {
+    NotificationsConfigMenu,
+    NotificationCard,
+  },
+  data() {
+    return {
+      show_config_menu: false,
+      show_old_messages: false,
+      timestamp: new Date(),
+      seconds_recent: 60,
+    }
+  },
+  computed: {
+    notifications(): CumulatedNotification[] {
+      return notification_store.active_cumulated_notifications
+    },
+    recent_notifications(): CumulatedNotification[] {
+      return this.notifications.filter(
+        (cumulated) => cumulated.notification.secondsTill(new Date()) < this.seconds_recent,
+      )
+    },
+    notifications_to_show(): CumulatedNotification[] {
+      const notifications_to_show = this.show_old_messages ? this.notifications : this.recent_notifications
+      // Returning the notifications reversed to show the last ones first
+      return notifications_to_show.slice().reverse()
+    },
+  },
+  mounted() {
+    setInterval(this.updateTimestamp, 1000)
```

Comment:
there's no `clearInterval`anywhere.

Reply:
@Williangalvani, since this component will only exist in our top bar and will only exist one, does it still have a problem ?
That does not invalidate the point that we should add a RAII here to avoid problem.

Reply:
I thought of that. it isn't an issue right now, it is just a matter of future-proofing things.